### PR TITLE
[CBRD-20541] retrieve PSTAT_PEEK_SINGLE_VALUE typed performance statistics

### PR DESCRIPTION
**Pull request by asturzu**
http://jira.cubrid.org/browse/CBRD-20541

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/session/session.c (1 incorrect indentation(s) found)
```
  2587 
```
#### File: src/storage/heap_file.c (1 incorrect indentation(s) found)
```
  1102 
```
#### File: src/thread/thread.c (3 incorrect indentation(s) found)
```
  3197 
  3223 
  3247 
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by eseokoh** [comment]
@asturzu 

There are other peeked type statistics:
```C
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_FIXED_CNT, "Num_data_page_fixed"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_DIRTY_CNT, "Num_data_page_dirty"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_LRU1_CNT, "Num_data_page_lru1"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_LRU2_CNT, "Num_data_page_lru2"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_AIN_CNT, "Num_data_page_ain"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_AVOID_DEALLOC_CNT, "Num_data_page_avoid_dealloc"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_AVOID_VICTIM_CNT, "Num_data_page_avoid_victim"),
  PSTAT_METADATA_INIT_SINGLE_PEEK (PSTAT_PB_VICTIM_CAND_CNT, "Num_data_page_victim_cand"),
```

Are they all right?

File: src/base/perf_monitor.c [hunk 1]
```diff
@@ -2238,9 +2245,21 @@ perfmon_server_dump_stats_to_buffer (const UINT64 * stats, char *buffer, int buf
 	    {
 	      if (pstat_Metadata[i].valtype != PSTAT_COUNTER_TIMER_VALUE)
```
Comment (eseokoh) [inline src/base/perf_monitor.c]: Please consider to change the line if you include all peeked type stats. 
I prefer to something like this:
```c
if ( ... == PSTAT_PEEK_SINGLE_VALUE)
  {
    ...
  }
else if ( != PSTAT_COUNTER_TIMER_VALUE)
  {
    ...
  }
```

Then, you don't need `get_value_from_stat`.

**Comment by asturzu** [comment]
Dear Mr Oh,

I used those 4 variables because they were kind of special cases. The rest of the peek statistics are 
computed in the perfmon_server_calc_stats function using the pgbuf_peek_stats function. 

I could use the global_stats array from pstat_Global instead of registering a duplicate array
for some of the peek stats.  It would be the same thing as it is now just that I would eliminate
those global variables.

Another solution would be to eliminate the  if (pstat_Global.is_watching[tran_index])
from perfmon_set_at_offset and use perfmon_set_stat as before. This would give more accurate results,
but would mean that you would be incrementing both the global and the private array kept for each thread in situations when there are no active threads.

**Comment by eseokoh** [comment]
@asturzu 

I think your approach is better than alternatives. 
One of my concerns is that a developer may wonder how to add a "peeked" value in the future.
I think we need comments to explain your approach. Please add comments on `get_value_from_stat` and declarations of special variables. 

Thank you :)

**Pull request merged**